sdk/storage/azure-storage-file-datalake/azure/storage/file/datalake/_data_lake_service_client.py

```diff
@@ -0,0 +1,371 @@
+        """
+        account_url, secondary, credential = parse_connection_str(conn_str, credential, 'dfs')
+        if 'secondary_hostname' not in kwargs:
+            kwargs['secondary_hostname'] = secondary
+        return cls(account_url, credential=credential, **kwargs)
+
+    def get_user_delegation_key(self, key_start_time,  # type: datetime
+                                key_expiry_time,  # type: datetime
+                                **kwargs  # type: Any
+                                ):
+        # type: (...) -> UserDelegationKey
+        """
+        Obtain a user delegation key for the purpose of signing SAS tokens.
+        A token credential must be present on the service object for this request to succeed.
+
+        :param ~datetime.datetime key_start_time:
+            A DateTime value. Indicates when the key becomes valid.
+        :param ~datetime.datetime key_expiry_time:
+            A DateTime value. Indicates when the key stops being valid.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :return: The user delegation key.
+        :rtype: ~azure.storage.file.datalake.UserDelegationKey
+        """
+        delegation_key = self._blob_service_client.get_user_delegation_key(key_start_time=key_start_time,
+                                                                           key_expiry_time=key_expiry_time,
+                                                                           **kwargs)
+        delegation_key._class_ = UserDelegationKey
+        return delegation_key
+
+    def list_file_systems(self, name_starts_with=None,  # type: Optional[str]
+                          include_metadata=None,  # type: Optional[bool]
+                          **kwargs):
+        # type: (...) -> ItemPaged[FileSystemProperties]
+        """Returns a generator to list the file systems under the specified account.
+
+        The generator will lazily follow the continuation tokens returned by
+        the service and stop when all file systems have been returned.
+
+        :param str name_starts_with:
+            Filters the results to return only file systems whose names
+            begin with the specified prefix.
+        :param bool include_metadata:
+            Specifies that file system metadata be returned in the response.
+            The default value is `False`.
+        :keyword int results_per_page:
+            The maximum number of file system names to retrieve per API
+            call. If the request does not specify the server will return up to 5,000 items per page.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :returns: An iterable (auto-paging) of FileSystemProperties.
+        :rtype: ~azure.core.paging.ItemPaged[~azure.storage.file.datalake.FileSystemProperties]
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/test_datalake_service_samples.py
```

Comment:
>../samples/test_datalake_service_samples.py [](start = 32, length = 43)

rm these non-existing sample references?

Reply:
Can we leave it there and add later?

Reply:
wouldn't they screw with the doc tool? since it's pointing to non-existing files 

if it works fine,then I'm ok with keeping them

---
In reply to: [340894009](https://github.com/Azure/azure-sdk-for-python/pull/8280#discussion_r340894009) [](ancestors = 340894009)